File: Algorithms/0598_Range_Addition_II/Java/RangeAdditionIISolution1.java

// Space: O(1)
// Time: O(n)

class Solution {
    public int maxCount(int m, int n, int[][] ops) {
        if (ops.length == 0) return m * n;
        int row = ops[0][0], column = ops[0][1];

        for (int[] item : ops) {
            row = Math.min(row, item[0]);
            column = Math.min(column, item[1]);
        }
        return row * column;
    }
}

File: Algorithms/0598_Range_Addition_II/Python/Range_Addition_II_Solution_1.py

# Space: O(n)
# Time: O(n)

class Solution:
    def maxCount(self, m, n, ops):
        if not ops: return m * n
        max_row, max_column = ops[0][0], ops[0][1]
        for i, j in ops[1:]:
            max_row = min(max_row, i)
            max_column = min(max_column, j)
        return max_row * max_column





